File: backoffice/src/shared/domain/exceptions/DomainException.py
"""
 *
 * Class
 *
"""

class DomainException( Exception ):

    """
     *
     * Attributes 
     *
    """

    __code : int

    """
     *
     * Methods 
     *
    """

    def __init__( self, code : int, message : str ) -> object:
        super().__init__( message )
        self.__code = code

    def getCode( self ) -> int:
        return self.__code

File: preparation/src/shared/domain/exceptions/DomainException.cs
/* 
 * 
 * Class
 *
*/

namespace preparation.src.shared.domain {

    public class DomainException : Exception {

        /* 
         * 
         * Attributes
         *
        */

        private int code;

        /* 
         * 
         * Methods
         *
        */

        public DomainException( int code, string message ) : base( message ) {
            this.code = code;
        }

        public int getCode() {
            return code;
        }

    }

}